### PR TITLE
add finindustries/tunnel75

### PR DESCRIPTION
**Pull request by Zengxlll**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/finindustries/tunnel75/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+LTO_ENABLE = yes
+
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/rules.mk:3]:
```suggestion
# This file intentionally left blank
```

File: keyboards/finindustries/tunnel75/tunnel75.c
Comment (zvecr) [inline keyboards/finindustries/tunnel75/tunnel75.c:1]: This file should be removed.

File: keyboards/finindustries/tunnel75/tunnel75_viav3(iso+7u).json
Comment (zvecr) [inline keyboards/finindustries/tunnel75/tunnel75_viav3(iso+7u).json:1]: This file should be removed.

File: keyboards/finindustries/tunnel75/tunnel75_viav3.json
Comment (zvecr) [inline keyboards/finindustries/tunnel75/tunnel75_viav3.json:1]: This file should be removed.

File: keyboards/finindustries/tunnel75/readme.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Tunnel75
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/readme.md:1]: This needs to better follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)

File: keyboards/finindustries/tunnel75/info.json [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+  "board": "STM32_F103_STM32DUINO",
+  "features": {
+    "bootmagic": true,
+    "command": false,
+    "console": false,
+    "extrakey": true,
+    "mousekey": true,
+    "nkro": true,
+    "rgblight": true,
+    "audio": false
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/info.json:22]:
```suggestion
    "rgblight": true
```

File: keyboards/finindustries/tunnel75/info.json [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+      "B11"
+    ],
+    "rows": ["B10", "A2", "A1", "A0", "B9", "B8"]
+  },
+  "diode_direction": "COL2ROW",
+  "debounce": 5,
+  "bootmagic": {
+    "matrix": [0, 0]
+  },
+
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/info.json:49]:
```suggestion
```

File: keyboards/finindustries/tunnel75/info.json [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+  "rgblight": {
+    "led_count": 12,
+    "pin": "B0",
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/info.json:52]: This is no longer a valid configuration, and is `ws2812.pin`
```suggestion
```

File: keyboards/finindustries/tunnel75/info.json [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+    "max_brightness": 255,
+    "sleep": true,
+    "hue_steps": 8,
+    "saturation_steps": 8,
+    "brightness_steps": 8,
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/info.json:57]:
```suggestion
    "sleep": true,
```

File: keyboards/finindustries/tunnel75/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    [3] = LAYOUT(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/keymaps/default/keymap.c:27]:
```suggestion
    [1] = LAYOUT(EE_CLR, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```

File: keyboards/finindustries/tunnel75/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    [3] = LAYOUT(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/keymaps/via/keymap.c:27]:
```suggestion
    [1] = LAYOUT(KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, EE_CLR, QK_BOOT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, NK_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, RGB_TOG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_SAI, KC_TRNS, KC_TRNS, KC_TRNS, RGB_HUI, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAI, KC_TRNS, KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS)
```

File: keyboards/finindustries/tunnel75/halconf.h
Comment (zvecr) [inline keyboards/finindustries/tunnel75/halconf.h:1]: You are not configuring pwm for anything, so this file should be removed.

File: keyboards/finindustries/tunnel75/mcuconf.h
Comment (zvecr) [inline keyboards/finindustries/tunnel75/mcuconf.h:1]: You are not configuring pwm for anything, so this file should be removed.

File: keyboards/finindustries/tunnel75/info.json [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+  "url": "",
+  "usb": {
+    "vid": "0xF1D1",
+    "pid": "0xF101",
+    "device_version": "0.0.1"
+  },
+  "processor": "STM32F103",
+  "bootloader": "uf2boot",
+  "board": "STM32_F103_STM32DUINO",
+  "features": {
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/info.json:14]:
```suggestion
    "build": {
        "lto": true
    },
  "features": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by zvecr** [comment]
Theres not much requesting review, if half the requested changes have not been completed.

File: keyboards/finindustries/tunnel75/config.h [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define FORCE_NKRO
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_MOOD
+#define RGB_DI_PIN B0
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/config.h:23]: Nope, `ws2812.pin` as previously mentioned.
```suggestion
```

File: keyboards/finindustries/tunnel75/readme.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+-   Keyboard Maintainer: [Zengxlll](https://github.com/Zengxlll)
+-   Hardware Supported: _The PCBs, controllers supported_
+-   Hardware Availability: _Links to where you can find this hardware_
```
Comment (zvecr) [inline keyboards/finindustries/tunnel75/readme.md:5]: This needs to be filled in, and not the template text.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/finindustries/tunnel75/rules.mk (hunk 1 above)
Comment (zvecr) [reply keyboards/finindustries/tunnel75/rules.mk]: This change has not been made correctly.

Comment (Zengxlll) [reply keyboards/finindustries/tunnel75/rules.mk]: How should I modify it?

Comment (zvecr) [reply keyboards/finindustries/tunnel75/rules.mk]: Like the suggestion.

Comment (Zengxlll) [reply keyboards/finindustries/tunnel75/rules.mk]: add this?    # This file intentionally left blank

**Comment by Zengxlll** [comment]
@zvecr Thank you for your review and suggestions.

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/finindustries/tunnel75/info.json [hunk 12]
```diff
@@ -0,0 +1,473 @@
+{
+    "keyboard_name": "Tunnel75",
+    "manufacturer": "FIN INDUSTRIES",
+    "maintainer": "FIN INDUSTRIES",
+    "url": "",
```
[the remark this answers is not in the document]
Comment (Zengxlll) [reply keyboards/finindustries/tunnel75/info.json:5]: Sorry, no.

File: keyboards/finindustries/tunnel75/info.json
[the remark this answers is not in the document]
Comment (Zengxlll) [reply keyboards/finindustries/tunnel75/info.json:1]: Sorry, I don't understand what you mean. Is the formatting inconsistent?

[the remark this answers is not in the document]
Comment (dunk2k) [reply keyboards/finindustries/tunnel75/info.json:1]: > Sorry, I don't understand what you mean. Is the formatting inconsistent?

Inconsistent when compared to other keyboard "layout" definitions, but functional none the less 🙂

Would you be able to change the formatting of each `"layout"` position to be consistent with the following format:
`{"matrix": [0, 1], "x": 0, "y": 0.3}`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/finindustries/tunnel75/info.json
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/finindustries/tunnel75/info.json:1]: `qmk format-json` should be used on the file. That should fix most of it.

File: keyboards/finindustries/tunnel75/keymaps/default/keymap.c
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/finindustries/tunnel75/keymaps/default/keymap.c:1]: Eg,  make each line representative of one row on the keyboard.  Also, most keycodes have a 7 character version, so they can/should be consistently spaced,  Eg `XXXXXXX, ` (7 chars, comma and a space).

While it's not quite the same, something like this: 
https://github.com/qmk/qmk_firmware/blob/master/layouts/default/75_ansi/default_75_ansi/keymap.c

Though, the ascii art isn't necessary.

**Pull request closed without merging**